### PR TITLE
Add GetPublished egghead state

### PR DESCRIPTION
**Pull request by trevordmiller**
- Break up GetPublished components into smaller pieces; I imagine a few of these will be promoted as more routes are created.
- Finish Get Published todo state for all egghead pieces; just missing FedEx + Slack
- Add tests for todo completion logic

fyi, this only works with the real endpoints currently since the `instructorLessons` logic uses the pagination setup. I'll see if I can seamlessly update the fakeApi to use it in the future; if not I'll have a ternary or something for that endpoint for the fakeApi.

batman

### CODE REVIEW

**Comment by trevordmiller** [comment]
Can't wait until this lands so we can consolidate lodash imports: https://github.com/lodash/lodash/issues/1457

File: src/App/screens/Instructor/screens/GetPublished/index.js [hunk 1]
```diff
@@ -1,100 +1,43 @@
...
+        main={
+          <GetPublishedSteps
+            instructor={instructor}
+            instructorLessons={instructorLessons}
+          />
+        }
+        aside={
+          <GetPublishedFaq />
+        }
+      />
```
Comment (trevordmiller) [inline src/App/screens/Instructor/screens/GetPublished/index.js:38]: I would imagine many of our routes might be able to use this `Split` component, depending on how we wireframe them

**Comment by joelhooks** [comment]
> Finish Get Published todo state for all egghead pieces; just missing FedEx + Slack

For Slack, we will save the instructors Slack Id on the instructor object, so if it exists, we will have that. I'm pretty close on this with the Slack integration, but you can assume that is the case.

We will a field the instructor that will contain the tracking number of their gear shipment. If it exists, this step is done.

We can solve fedex in several ways. My first thought is to use slack again, with a slash or bot command coupled with the fedex tracking number. Or, you coulkd say `@eggo instructor Jane Great` and the returned instructor object would contain a button that says "Verify Gear Sent" if you have the perfs and when pressed asks you to enter the number. Or both...!

Does that sound reasonable?

**Comment by trevordmiller** [comment]
@joelhooks Yeah sounds like a great idea. Let me know what you want those new `instructor` object properties to be called for the Slack ID and gear tracking number and I'll add them in another PR. Thanks!

File: src/App/screens/Instructor/screens/GetPublished/index.js (hunk 1 above)
Comment (joelhooks) [reply src/App/screens/Instructor/screens/GetPublished/index.js:38]: I agree, layouts like this are definitely candidates for promotion. Very useful

File: src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/utils/isStepComplete.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import includes from 'lodash/includes'
```
Comment (joelhooks) [inline src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/utils/isStepComplete.js:1]: I like this a lot. breaking the utilities out into well tested functions that can be imported and used as needed.

File: src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import DescriptionBlock from './components/DescriptionBlock'
+
+const GetPublishedSteps = ({
+  instructor,
+  instructorLessons,
+}) => {
+
+  const instructorLessonStates = compact(uniq(map(instructorLessons.lessons, 'state')))
+
+  const steps = [
```
Comment (joelhooks) [inline src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/index.js:16]: I think we will be able to do some cool things with loading in content dynamically as you suggested. The instructor guide needs some TLC in the editing department. Structure, consistency, filling in the gaps. This is very high on my todo list.

File: src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/utils/isStepComplete.js (hunk 2 above)
Comment (trevordmiller) [reply src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/utils/isStepComplete.js:1]: Thanks! Agreed 👍

File: src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/index.js (hunk 3 above)
Comment (trevordmiller) [reply src/App/screens/Instructor/screens/GetPublished/components/GetPublishedSteps/index.js:16]: 👏

**Pull request merged**